iceoryx_posh/source/roudi/process_manager.cpp

```diff
@@ -516,8 +495,22 @@ void ProcessManager::addPublisherForProcess(const RuntimeName_t& name,
     searchForProcessAndThen(
         name,
         [&](Process& process) { // create a PublisherPort
+            auto segmentInfo = m_segmentManager->getSegmentInformationWithWriteAccessForUser(process.getUser());
```

Comment:
Have you thought about having a `process.getSegmentInfoWithWriteAccess` instead of `process.getUser` followed with the call to get the information? This has the advantage that the segment info must not be queried again for each publisher of the process.

Reply:
This would be a possible optimisation. For me in this context it feels more natural to associate a `roudi::Process` with a `posix::User` instead of associating it with a `SegmentUserInformation`. Furthermore, I think the performance difference is neglectable. I would follow our normal mantra here: Make it work, make it beautiful, make it fast. Something for the next PR.